relay-server/src/processing/transactions/mod.rs

```diff
@@ -0,0 +1,595 @@
+        let quotas_client = None;
+        let sampling_result =
+            process::run_dynamic_sampling(&work, ctx, filters_status, quotas_client).await;
+
+        #[cfg(feature = "processing")]
+        let server_sample_rate = sampling_result.sample_rate();
+
+        if let Some(outcome) = sampling_result.into_dropped_outcome() {
+            let work = process::process_profile(work, ctx, SamplingDecision::Drop);
+            let (work, extracted_metrics) =
+                process::extract_metrics(work, ctx, SamplingDecision::Drop)?;
+
+            let headers = work.headers.clone();
+            let profile = process::drop_after_sampling(work, ctx, outcome);
+            let mut profile = profile.transpose();
+            if let Some(profile) = profile.as_mut() {
+                self.limiter.enforce_quotas(profile, ctx).await?;
+            }
+
+            return Ok(super::Output {
+                main: profile.map(|managed| {
+                    TransactionOutput::OnlyProfile(
+                        managed.map(|Profile(item), _| ProfileWithHeaders { headers, item }),
+                    )
+                }),
+                metrics: Some(extracted_metrics),
+            });
+        }
+
+        // Need to scrub the transaction before extracting spans.
+        work = process::scrub(work, ctx)?;
+
+        #[cfg(feature = "processing")]
+        if ctx.config.processing_enabled() {
+            // Process profiles before extracting metrics, to make sure they are removed if they are invalid.
+            let work = process::process_profile(work, ctx, SamplingDecision::Keep);
+
+            let (indexed, extracted_metrics) =
+                process::extract_metrics(work, ctx, SamplingDecision::Keep)?;
+
+            let mut indexed = process::extract_spans(indexed, ctx, server_sample_rate);
+
+            self.limiter.enforce_quotas(&mut indexed, ctx).await?;
+
+            if !indexed.flags.fully_normalized {
+                relay_log::error!(
+                    tags.project = %project_id,
+                    tags.ty = event_type(&indexed.transaction.0).map(|e| e.to_string()).unwrap_or("none".to_owned()),
+                    "ingested event without normalizing"
+                );
+            };
+
+            return Ok(super::Output {
+                main: Some(TransactionOutput::Indexed(indexed)),
+                metrics: Some(extracted_metrics),
+            });
+        }
+
+        self.limiter.enforce_quotas(&mut work, ctx).await?;
+
+        let metrics = work.wrap(extracted_metrics.into_inner());
+        Ok(super::Output {
+            main: Some(TransactionOutput::Full(work)),
+            metrics: Some(metrics),
+        })
+    }
+}
+
+/// A transaction in its serialized state, as transported in an envelope.
+#[derive(Debug)]
+pub struct SerializedTransaction {
+    headers: EnvelopeHeaders,
+    transaction: Item,
+    attachments: smallvec::SmallVec<[Item; 3]>,
```

Comment:
We have an `Items` type alias for this.


```suggestion
    attachments: Items,
```